instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/ui/routeline/AlternativeRouteSelectionTest.kt

```diff
@@ -0,0 +1,206 @@
+                    // only need one route progress for this test
+                    mapboxNavigation.unregisterRouteProgressObserver(this)
+
+                    assertEquals(routeLineApi.getRoutes()[0], routeProgress.route)
+                }
+            })
+            mapboxNavigation.startTripSession()
+        }
+
+        Espresso.onIdle()
+    }
+
+    fun addRouteLine() {
+        runOnMainSync {
+            routeLineView = MapboxRouteLineView(MapboxRouteLineOptions.Builder(activity).build())
+            routeLineApi = MapboxRouteLineApi(MapboxRouteLineOptions.Builder(activity).build())
+
+            mapboxNavigation.registerRoutesObserver(object : RoutesObserver {
+                override fun onRoutesChanged(routes: List<DirectionsRoute>) {
+                    val routeLines = directionsResponse.routes().map {
+                        RouteLine(it, null)
+                    }
+                    routeLineApi.setRoutes(routeLines).apply {
+                        routeLineView.render(activity.mapboxMap.getStyle()!!, this)
+                    }
+                    mapboxNavigation.unregisterRoutesObserver(this)
+                }
+            })
+        }
+    }
+
+    fun addNavigationCamera() {
+        runOnMainSync {
+            mapboxNavigationViewportDataSource = MapboxNavigationViewportDataSource(
+                MapboxNavigationViewportDataSourceOptions.Builder().build(),
+                activity.mapboxMap
+            )
+            navigationCamera = NavigationCamera(
+                activity.mapboxMap,
+                activity.mapView.getCameraAnimationsPlugin(),
+                mapboxNavigationViewportDataSource
+            )
+            navigationCamera.requestNavigationCameraToFollowing()
+
+            mapboxNavigation.registerRoutesObserver(object : RoutesObserver {
+                override fun onRoutesChanged(routes: List<DirectionsRoute>) {
+                    mapboxNavigationViewportDataSource.onRouteChanged(routes.first())
+                    mapboxNavigationViewportDataSource.evaluate()
+                }
+            })
+
+            mapboxNavigation.registerRouteProgressObserver(object : RouteProgressObserver {
+                override fun onRouteProgressChanged(routeProgress: RouteProgress) {
+                    mapboxNavigationViewportDataSource.onRouteProgressChanged(routeProgress)
+                    mapboxNavigationViewportDataSource.evaluate()
+                }
+            })
+
+            mapboxNavigation.registerMapMatcherResultObserver(object : MapMatcherResultObserver {
+                override fun onNewMapMatcherResult(mapMatcherResult: MapMatcherResult) {
+                    mapboxNavigationViewportDataSource.onLocationChanged(
+                        mapMatcherResult.enhancedLocation
+                    )
+                    mapboxNavigationViewportDataSource.evaluate()
+                }
+            })
+        }
+    }
+
+    fun addLocationPuck() {
```

Comment:
We should be able to remove this, just to leave the test to a bare minimum.

Reply:
removed